### PR TITLE
Remove unnecessary  check for negative bestChainHeight

### PR DESCRIPTION
**Pull request by pmprete**
This was detected by @yeousunn at #697, thanks for your colaboration.
Also add tests for setLockWhitelistDisableBlockDelay and some refactoring.

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 1]
```diff
@@ -1822,17 +1822,21 @@ public Integer setLockWhitelistDisableBlockDelay(Transaction tx, BigInteger disa
...
         if (lockWhitelist.isDisableBlockSet()) {
             return -1;
         }
-        int disableBlockDelay = disableBlockDelayBI.intValueExact();
-        int bestChainHeight = getBtcBlockchainBestChainHeight();
-        if (bestChainHeight < 0 || Integer.MAX_VALUE - disableBlockDelay < bestChainHeight) {
+        try {
+            int disableBlockDelay = disableBlockDelayBI.intValueExact();
+            int bestChainHeight = getBtcBlockchainBestChainHeight();
+            if (bestChainHeight < 0 || disableBlockDelay < 0 ) {
```
Comment (amendelzon) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: Why would `bestChainHeight` be negative?

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java [hunk 2]
```diff
@@ -2925,6 +2925,182 @@ public void addLockWhitelistAddress_invalidAddress() throws IOException {
...
+        BridgeSupport bridgeSupport = getBridgeSupportWithMocksAndBtcBlockstoreForWhitelistTests(mockedWhitelist,btcBlockStore);
+
+        BigInteger disableBlockDelayBI = BigInteger.valueOf(Integer.MAX_VALUE / 2);
+
+        Assert.assertEquals(-2, bridgeSupport.setLockWhitelistDisableBlockDelay(mockedTx, disableBlockDelayBI).intValue());
+        verify(mockedWhitelist, never()).put(any(), any());
+    }
+
+    @Test
+    public void setLockWhitelistDisableBlockDelay_maxValueDisableBlockBI() throws IOException, BlockStoreException {
```
Comment (amendelzon) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java]: This is a duplicated test case I think.

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java [hunk 3]
```diff
@@ -2925,6 +2925,182 @@ public void addLockWhitelistAddress_invalidAddress() throws IOException {
...
+        BridgeSupport bridgeSupport = getBridgeSupportWithMocksAndBtcBlockstoreForWhitelistTests(mockedWhitelist,btcBlockStore);
+
+        BigInteger disableBlockDelayBI = BigInteger.valueOf(Integer.MAX_VALUE);
+
+        Assert.assertEquals(-2, bridgeSupport.setLockWhitelistDisableBlockDelay(mockedTx, disableBlockDelayBI).intValue());
+        verify(mockedWhitelist, never()).put(any(), any());
+    }
+
+    @Test
+    public void setLockWhitelistDisableBlockDelay_negativeBestChain() throws IOException, BlockStoreException {
```
Comment (amendelzon) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java]: I don't think we need to cover this case.

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java (hunk 2 above)
Comment (pmprete) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java]: one is the max int value, the other one is the max long value.
The max long test the exception given when converting from long number, the max int value test the edge case

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java (hunk 1 above)
Comment (pmprete) [reply rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java]: my bad, i thougth that #697 was merged already

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java (hunk 3 above)
Comment (pmprete) [reply rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java]: good catch

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java [hunk 4]
```diff
@@ -2925,6 +2925,158 @@ public void addLockWhitelistAddress_invalidAddress() throws IOException {
...
+        LockWhitelist mockedWhitelist = mock(LockWhitelist.class);
+        when(mockedWhitelist.isDisableBlockSet()).thenReturn(false);
+        int bestChainHeight = (Integer.MAX_VALUE / 2) + 2;
+        BtcBlockstoreWithCache btcBlockStore = mock(BtcBlockstoreWithCache.class);
+        StoredBlock storedBlock = mock(StoredBlock.class);
+        when(storedBlock.getHeight()).thenReturn(bestChainHeight);
+        when(btcBlockStore.getChainHead()).thenReturn(storedBlock);
+        BridgeSupport bridgeSupport = getBridgeSupportWithMocksAndBtcBlockstoreForWhitelistTests(mockedWhitelist,btcBlockStore);
+
+        BigInteger disableBlockDelayBI = BigInteger.valueOf(Integer.MAX_VALUE / 2);
```
Comment (juli) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java]: If you want to test the case when current block height + delay is negative then use `Integer.MAX_VALUE`  otherwise it won't be negative until the Bitcoin block height is `Integer.MAX_VALUE / 2 + 1` and that will be in 40,000 years.
